### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by miekg**
This kicks of a "release" whenever a release-* tag is pushed.

Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./systemk
+          asset_name: systemk
+          asset_content_type: application/octet-stream
+
+      - name: Delete Older Releases
+        uses: dev-drprasad/delete-older-releases@v0.1.0
```
Comment (pires) [inline .github/workflows/release.yml:49]: Why would we delete releases?

**Review by pires — COMMENTED** [review]
Overall LGTM minus deleting releases.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [virtual-kubelet/systemk] Add r..." ]
>@pires commented on this pull request.
>
>Overall LGTM minus deleting releases.
>
>━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
>
>In .github/workflows/release.yml:
>
>> +          release_name: Release-${{ env.EPOCH }}
>+          draft: false
>+          prerelease: false
>+
>+      - name: Upload Release Asset
>+        uses: actions/upload-release-asset@v1
>+        env:
>+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
>+        with:
>+          upload_url: ${{ steps.create_release.outputs.upload_url }}
>+          asset_path: ./systemk
>+          asset_name: systemk
>+          asset_content_type: application/octet-stream
>+
>+      - name: Delete Older Releases
>+        uses: dev-drprasad/delete-older-releases@v0.1.0
>
>Why would we delete releases?

maybe not needed here... I another project I do releases on every commit, which may lead
to an excessive amount of releases.

**Comment by pires** [comment]
I think it would be good to follow an approach like what you do for CoreDNS where a release is made when there's enough justification for a release, hence not delete releases.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [virtual-kubelet/systemk] Add r..." ]
>I think it would be good to follow an approach like what you do for CoreDNS
>where a release is made when there's enough justification for a release, hence
>not delete releases.

yeah, we don't need that here.

**Review by pires — APPROVED** [review]

**Pull request merged**